core/src/main/java/apoc/cypher/Cypher.java

```diff
@@ -81,103 +80,46 @@ public Stream<MapResult> run(@Name("statement") String statement, @Name("params"
-        while (scanner.hasNext()) {
-            String stmt = removeShellControlCommands(scanner.next());
-            if (stmt.trim().isEmpty()) continue;
-            if (isSchemaOperation(stmt)) {
-                Util.inTx(db, pools, txInThread -> {
-                    try (Result result = txInThread.execute(stmt, params)) {
-                        return consumeResult(result, queue, addStatistics);
-                    }
-                });
-            }
-        }
-    }
-
     @Procedure(name = "apoc.cypher.runMany", mode = WRITE)
     @Description("Runs each semicolon separated statement and returns a summary of the statement outcomes.")
     public Stream<RowResult> runMany(
             @Name("statement") String cypher,
             @Name("params") Map<String, Object> params,
             @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
         boolean addStatistics = Util.toBoolean(config.getOrDefault("statistics", true));
-        int queueCapacity = Util.toInteger(config.getOrDefault("queueCapacity", 100));
 
-        StringReader stringReader = new StringReader(cypher);
-        return runManyStatements(stringReader, params, false, addStatistics, queueCapacity);
+        return Iterators.stream(new Scanner(new StringReader(cypher)).useDelimiter(";\r?\n"))
+                .map(Cypher::removeShellControlCommands)
+                .filter(s -> !s.isBlank())
+                .flatMap(s -> streamInNewTx(s, params, addStatistics));
+    }
+
+    private Stream<Cypher.RowResult> streamInNewTx(String cypher, Map<String, Object> params, boolean stats) {
+        final var innerTx = db.beginTx();
+        try {
+            // Hello fellow wanderer,
+            // At this point you may have questions like;
+            // - "Why do we execute this statement in a new transaction?"
+            // My guess is as good as yours. This is the way of the apoc. Safe travels.
+            final var results = new RunManyResultSpliterator(innerTx.execute(cypher, params), stats);
+            return StreamSupport.stream(results, false).onClose(results::close).onClose(innerTx::commit);
+        } catch (AuthorizationViolationException accessModeException) {
+            // We meet again, few people make it this far into this world!
+            // I hope you're not still seeking answers, there are few to give.
+            // It has been written, in some long forgotten commits,
+            // that failures of this kind should be avoided. The ancestors
+            // were brave and used a regex based cypher parser to avoid
+            // trying to execute schema changing statements all together.
+            // We don't have that courage, and try to forget about it
+            // after the fact instead.
+            // One can only hope that by keeping this tradition alive,
+            // in some form, we make some poor souls happier.
+            innerTx.close();
+            return Stream.empty();
```

Comment:
I was very tempted to not include this, but decided to anyway, many tests relies on this.

Reply:
Sounds good, I'll keep it here to not break tests and create a card to investigate if this is really what we want to do.

Reply:
https://trello.com/c/5iHK0d0X